spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/KubernetesConfigDataLocationResolver.java

```diff
@@ -0,0 +1,223 @@
+		bootstrapContext.addCloseListener(event -> event.getApplicationContext().getBeanFactory().registerSingleton(
+				"configDataSecretsConfigProperties", event.getBootstrapContext().get(SecretsConfigProperties.class)));
+
+		HashMap<String, Object> kubernetesConfigData = new HashMap<>();
+		kubernetesConfigData.put("spring.cloud.kubernetes.client.namespace", properties.getNamespace());
+		if (propertyHolder.applicationName != null) {
+			// If its null it means sprig.application.name was not set so don't add it to
+			// the property source
+			kubernetesConfigData.put("spring.application.name", propertyHolder.applicationName);
+		}
+		PropertySource<Map<String, Object>> propertySource = new MapPropertySource("kubernetesConfigData",
+				kubernetesConfigData);
+		ConfigurableEnvironment environment = new StandardEnvironment();
+		environment.getPropertySources().addLast(propertySource);
+		environment.setActiveProfiles(profiles.getAccepted().toArray(new String[profiles.getAccepted().size()]));
+		KubernetesNamespaceProvider namespaceProvider = kubernetesNamespaceProvider(environment);
+
+		registerBeans(resolverContext, location, profiles, propertyHolder, namespaceProvider);
+
+		KubernetesConfigDataResource resource = new KubernetesConfigDataResource(properties, configMapProperties,
+				secretsProperties, location.isOptional(), profiles, environment);
+		resource.setLog(log);
+
+		List<KubernetesConfigDataResource> locations = new ArrayList<>();
+		locations.add(resource);
+
+		return locations;
+	}
+
+	protected abstract void registerBeans(ConfigDataLocationResolverContext resolverContext,
+			ConfigDataLocation location, Profiles profiles, PropertyHolder propertyHolder,
+			KubernetesNamespaceProvider namespaceProvider);
+
+	protected boolean isRetryEnabled(ConfigMapConfigProperties configMapProperties,
+			SecretsConfigProperties secretsProperties) {
+		return isRetryEnabledForConfigMap(configMapProperties) || isRetryEnabledForSecrets(secretsProperties);
+	}
+
+	protected boolean isRetryEnabledForConfigMap(ConfigMapConfigProperties configMapProperties) {
+		return RETRY_IS_PRESENT && configMapProperties.getRetry().isEnabled() && configMapProperties.isFailFast();
+	}
+
+	protected boolean isRetryEnabledForSecrets(SecretsConfigProperties secretsProperties) {
+		return RETRY_IS_PRESENT && secretsProperties.getRetry().isEnabled() && secretsProperties.isFailFast();
+	}
+
+	protected KubernetesNamespaceProvider kubernetesNamespaceProvider(Environment environment) {
+		return new KubernetesNamespaceProvider(environment);
+	}
+
+	@Override
+	public int getOrder() {
+		return -1;
+	}
+
+	private BindHandler getBindHandler(ConfigDataLocationResolverContext context) {
+		return context.getBootstrapContext().getOrElse(BindHandler.class, null);
+	}
+
+	protected PropertyHolder loadProperties(ConfigDataLocationResolverContext context, String uris) {
```

Comment:
`uris` is not used within the method, so is `configMapEnabled` and `secretsEnabled`. Did you leave them here because they might be in a future? Or simply an artifact of trying various coding things?

Reply:
No just me not cleaning up my code lol